packages/contracts-bedrock/src/safe/TimelockGuard.sol

```diff
@@ -0,0 +1,233 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+// Safe
+import { GnosisSafe as Safe } from "safe-contracts/GnosisSafe.sol";
+import { Enum } from "safe-contracts/common/Enum.sol";
+import { GuardManager, Guard as IGuard } from "safe-contracts/base/GuardManager.sol";
+
+// Interfaces
+import { ISemver } from "interfaces/universal/ISemver.sol";
+
+/// @title TimelockGuard
+/// @notice This guard provides timelock functionality for Safe transactions
+/// @dev This is a singleton contract. To use it:
+///      1. The Safe must first enable this guard using GuardManager.setGuard()
+///      2. The Safe must then configure the guard by calling configureTimelockGuard()
+contract TimelockGuard is IGuard, ISemver {
+    /// @notice Configuration for a Safe's timelock guard
+    struct GuardConfig {
+        uint256 timelockDelay;
+    }
+
+    /// @notice Mapping from Safe address to its guard configuration
+    mapping(address => GuardConfig) public safeConfigs;
+
+    /// @notice Mapping from Safe address to its current cancellation threshold
+    mapping(address => uint256) public safeCancellationThreshold;
+
+    /// @notice Error for when guard is not enabled for the Safe
+    error TimelockGuard_GuardNotEnabled();
+
+    /// @notice Error for when Safe is not configured for this guard
+    error TimelockGuard_GuardNotConfigured();
+
+    /// @notice Error for when attempt to clear guard while it is still enabled for the Safe
+    error TimelockGuard_GuardStillEnabled();
+
+    /// @notice Error for invalid timelock delay
+    error TimelockGuard_InvalidTimelockDelay();
+
+    /// @notice Emitted when a Safe configures the guard
+    event GuardConfigured(address indexed safe, uint256 timelockDelay);
+
+    /// @notice Emitted when a Safe clears the guard configuration
+    event GuardCleared(address indexed safe);
+
+    /// @notice Semantic version.
+    /// @custom:semver 1.0.0
+    string public constant version = "1.0.0";
+
+    /// @notice Returns the timelock delay for a given Safe
+    /// @dev MUST never revert
+    /// @param _safe The Safe address to query
+    /// @return The timelock delay in seconds
+    function viewTimelockGuardConfiguration(address _safe) public view returns (uint256) {
+        // Q: What should this return if the guard is not enabled?
```

Comment:
Discussion on the spec: https://github.com/ethereum-optimism/specs/pull/761#discussion_r2333751553

Reply:
Added here: 92b33baca112, which is in this PR: https://github.com/ethereum-optimism/optimism/pull/17465